wf-galleon/EAP7-1217_Galleon_User_FP_Builder.adoc

```diff
@@ -0,0 +1,192 @@
+goal named _build-user-feature-pack_.
+This goal handles part of the complexity one would be faced when building a feature-pack for WildFly. 
+With minimal configuration, a user should be able to package his content into a WildFly compliant feature-pack.
+
+=== Possible usages inside an S2I context
+
+* Pre-built and already deployed custom feature-packs would be installed like WildFly (galleon install or provision).
+* The application src project could include a well-known directory (eg: _wildfly-feature-pack_) that would be recognized by WidlFly S2I. This
+directory would contain feature-pack(s) to build and install during application build process. 
+Calling _mvn clean install_ in this directory would build the feature-pack(s) and install it(them) into the local maven repository. 
+Ready to be provisioned by Galleon in a next step.
+
+=== Supported Galleon content
+Content present in user feature-packs is a subset of what Galleon defines. We are reducing the support to:
+
+* Ability to define JBoss Module modules. Modules are recognized by the plugin, 
+translated onto Galleon _packages_ and ready to be installed into 
+the right location. NB: Custom modules are installed directly under <wildfly-home>/modules/ dir.
+Custom modules will benefit from _thin-server_ and _fat-server_ Galleon installation capability. 
+Furthermore, the content of module.xml is traversed
+and guarantee is made that all non optional dependencies are installed with the custom module.
+* Ability to define _package_. Packages contain content to be installed inside a given location. 
+Package will benefit from tasks execution.
+* Ability to define _layer_. Layers are an interesting way to define re-usable content (eg: jdbc driver).
+* Ability to define _configuration_. Configurations are expected to contain piece of 
+configuration (eg: datasource, ...) to be combined with WildFly
+server configuration.
+
+=== Defining the content
+The following layout is expected in the resources dir (dir location configurable):
+
+* _configs_/<model>/<name>/config.xml
+* _layers_/<config model>/<name>/layer-spec.xml
+* _modules_/<module path>/main/module.xml (and any needed resources)
+* _packages_/<name>/package.xml +
+_packages_/<name>/[content/<content> | pm/wildfly/tasks.xml]
+
+=== Feature-pack configuration
+
+==== File based configuration
+If a file named _wildfly-user-feature-pack-build.xml_ is present in the project root dir (file 
+name and dir location configurable), it is used as the source of configuration. This file has the same content
+as _wildfly-feature-pack-build.xml_ (file used to build WildFly feature-pack).
+
+The following items are recognized:
+
+* Built feature-pack FPL (Feature Pack Location).
+* Dependencies on other feature-packs.
+* Default packages.
+
+Because configuration can be seen in some cases as over-complex and error-prone, 
+we propose to generate a default configuration when no configuration file is present.
```

Comment:
Yep, community documentation is needed. I mean over-complex for simple cases, otherwise when you have specific requirements, the configuration file is just fine.